### PR TITLE
Use any key to exit a blank slide

### LINKED ISSUE #20: Any-key should exit blank-slide mode
**Describe the bug**
Currently, blank slide mode sticks around until it is explicitly toggled.

**To Reproduce**
Steps to reproduce the behaviour:
1. Open the demo
2. Toggle blank slide mode with `.` or `,`
3. Hit an arrow key (for example)
4. See that scroll position moves, but the blank slide is still present.

**Expected behaviour**
We should exit out of blank slide mode.

### PR DESCRIPTION
**Pull request by mirisuzanne**
## Related Issue(s)

Fixes #20 

## Steps to test/reproduce

- Hit `.` or `,` to enter blank screen
- Hit any key to exit, without any other impact on the presentation

### CODE REVIEW

**Review by jamesnw — CHANGES_REQUESTED** [review]
This works when the blank slide is triggered with a shortcut, but not when it's triggered with a slide-event button- `<button slide-event="blankSlide">blankSlide</button>`.

In that case, the button retains focus, and since our keyup event is attached to the body, and the button stops propagation of the event, the #keyEventActions is never run. The user needs to click on the blank slide or re-trigger the button with Enter or Space bar.

I'm not sure what the solution is here- could we move the focus to the body, or to the blank slide (although that's an ::after pseudoelement), when the blank slide is opened?

**Comment by mirisuzanne** [comment]
@jamesnw what if we use a blank-slide dialog in the shadow-dom with a close button inside it? Then the blank slide would go in the top-layer, and move focus into the dialog, and we could add the special event listener there? Similar to our special control-panel-exiting listener?

**Comment by jamesnw** [comment]
That sounds like a good solution to me. What styles would the close button have? For this purpose, I'm thinking it could make sense to have the button be the white/black full screen element itself, with no text?

File: slide-deck.js [hunk 1]
```diff
@@ -43,11 +45,62 @@ class slideDeck extends HTMLElement {
...
+          </form>
+        </dialog>
+      </slot>
     `;
     const shadowRoot = node.attachShadow({ mode: "open" });
     shadowRoot.appendChild(template.content.cloneNode(true));
   }
 
+  // css
+  static adoptShadowStyles = (node) => {
```
Comment (jamesnw) [inline slide-deck.js:61]: Should this be private?

**Review by jamesnw — APPROVED** [review]
This works really well! 

I added an example of a slotted blank slide, and was initially surprised that the full screen styles weren't translated over. My recommendation would be to add a note that the styles in `adoptShadowStyles` are a good starting point, but it makes sense to have it be a blank slate.

**Comment by mirisuzanne** [comment]
@jamesnw I made some updates here based on your notes:

- `adoptShadowStyles` is now private
- I went ahead and applied only the full-screen styles to the slotted blank, I think your first instinct was right there. The shadow styles will still just be defaults, easy enough to override from the light dom
- While I was at it, I simplified and organized the styles some. There was redundant stuff in there. And I figured the little top-right 'x' is enough to avoid any exit confusion, without being distracting.

Maybe @stacyk or @dvdherron wants to review those styles as well?

**Review by stacyk — COMMENTED** [review]
To confirm, the blank screen keyboard shortcuts do not work the same in the custom control panel example. They both go to a blank white screen, but never the blank black screen. I wasn't sure if that is intentional?

You can see the left border on the custom example blank screen. Not sure if we want to add/move the `border: 0` values to  

```
[part=blank-slide],
::slotted([slot=blank-slide]) 
```

Screenshot 2024-01-15 at 12 22 35 PM

File: slide-deck.js [hunk 2]
```diff
@@ -43,11 +45,57 @@ class slideDeck extends HTMLElement {
...
+
+        &[open] { display: grid; }
+        form { display: grid; }
+      }
+
+      :host([blank-slide=white]) {
+        --sd-blank-slide-scheme: light;
+      }
+
+      [part=close-blank-slide] {
```
[the remark this answers is not in the document]
Comment (mirisuzanne) [reply slide-deck.js:87]: I don't know. Changing the cursor on buttons is contentious. I like it, but I've heard a11y folks say it's incorrect.

**Comment by mirisuzanne** [comment]
@stacyk I applied the border and color scheme to the slotted element. Does that look better?

File: slide-deck.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (stacyk) [reply slide-deck.js:87]: I like it too, but since pointer is for links, then we shouldn't rock the boat.

**Pull request merged**